app/components/task/extension-form.js

```diff
@@ -0,0 +1,124 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { resource, use } from 'ember-resources';
+import { TrackedMap } from 'tracked-maps-and-sets';
+import ENV from 'website-my/config/environment';
+import { action } from '@ember/object';
+import { WARNING_INVALID_NEW_ETA } from '../../constants/user-status';
+import { toastNotificationTimeoutOptions } from '../../constants/toast-notification';
+import { inject as service } from '@ember/service';
+
+export default class ExtensionFormComponent extends Component {
+  @tracked createExtensionRequest = false;
+  @service toast;
+  @service userState;
+
+  oldETA = new Date(this.args.task.endsOn * 1000).toLocaleString();
+
+  @use load = resource(({ on }) => {
+    const state = new TrackedMap();
+    const controller = new AbortController();
+
+    on.cleanup(() => controller.abort());
+    (async () => {
+      if (this.args.task) {
+        state.set('isLoading', true);
+        try {
+          const response = await fetch(
+            `${ENV.BASE_API_URL}/extensionRequests/self/?taskId=${this.args.task.id}`,
+            {
+              credentials: 'include',
+              signal: controller.signal,
+            }
+          );
+          if (response.status !== 200) {
+            this.toast.error(
```

Comment:
Toast stays for an infinite amount of time until we manually put close the toast.
Can we please have a timeout here?
![image](https://user-images.githubusercontent.com/78433013/213880843-02f2ad8d-ffaa-4c37-9e5e-fb3064a01853.png)